src/Illuminate/Support/Str.php

```diff
@@ -341,6 +341,39 @@ public static function convertCase(string $string, int $mode = MB_CASE_FOLD, ?st
         return mb_convert_case($string, $mode, $encoding);
     }
 
+    /**
+     * Compresses a string using gzip compression.
+     *
+     * @param  string  $string  The string to compress.
+     * @param  int  $mode  Compression level (1 to 9, defaults to 5).
+     * @return string|false The compressed string, or false on failure.
+     */
+    public static function compress(string $string, int $mode = 5)
+    {
+        if ($mode < 1 || $mode > 9) {
+            throw new \OutOfBoundsException('Compression mode must be between 1 and 9, default 5.');
+        }
+
+        return gzcompress($string, $mode);
```

Comment:
As `decompress` checks for failure (returning `false`), I guess, for symmetry, there should be such a check here too.